twist_pid_controller:
  ros__parameters:
    kp_linear_x: 1.0
    ki_linear_x: 0.0
    kd_linear_x: 0.0

    kp_linear_y: 0.0
    ki_linear_y: 0.0
    kd_linear_y: 0.0

    kp_linear_z: 0.0
    ki_linear_z: 0.0
    kd_linear_z: 0.0

    kp_angular_x: 0.0
    ki_angular_x: 0.0
    kd_angular_x: 0.0

    kp_angular_y: 0.0
    ki_angular_y: 0.0
    kd_angular_y: 0.0

    kp_angular_z: 1.0
    ki_angular_z: 0.0
    kd_angular_z: 0.0

    feed_forward: true

    max_integral_linear: 1.0
    max_integral_angular: 1.0

    control_frequency: 100.0

    timeout: 0.5

    debug: true  # Enable debug mode
    debug_topic: "pid_debug"  # Topic name for debug messages
    joy_topic: "joy"  # Topic name for the joystick

    # Topic name for the command velocity
    cmd_vel_input_topic: "/nav_vel_cmd"
    feedback_vel_topic: "/local_odometry"
    cmd_vel_out_topic: "/nav_vel"

    k_increment: 0.05  # Increment for the PID gains

    

